internal/reader/rmn_home_test.go

```diff
@@ -53,80 +54,97 @@ func TestRMNHomeChainConfigPoller_Ready(t *testing.T) {
 	require.NoError(t, configPoller.Close())
 }
 
-func TestRMNHomeChainConfigPoller_HealthReport(t *testing.T) {
-	homeChainReader := readermock.NewMockContractReaderFacade(t)
+func TestRMNHomePoller_HealthReport(t *testing.T) {
+	t.Parallel()
 
-	var (
-		tickTime       = 1 * time.Millisecond
-		totalSleepTime = 50 * time.Millisecond // give more time for multiple ticks
-	)
+	tests := []struct {
+		name        string
+		failedPolls uint
+		wantErr     bool
+	}{
+		{
+			name:        "Healthy state",
+			failedPolls: 0,
+			wantErr:     false,
+		},
+		{
+			name:        "Unhealthy state",
+			failedPolls: MaxFailedPolls,
+			wantErr:     true,
+		},
+	}
 
-	// Set up the mock to return an error for the first 10 calls
-	homeChainReader.On(
-		"GetLatestValue",
-		mock.Anything,
-		mock.Anything,
-		mock.Anything,
-		mock.Anything,
-		mock.Anything,
-	).Return(fmt.Errorf("error")).Times(50)
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			homeChainReader := readermock.NewMockContractReaderFacade(t)
 
-	// Set up the mock for a successful call after 10 errors
-	primaryConfig, secondaryConfig := createTestRMNHomeConfigs(false, false)
-	rmnHomeOnChainConfigs := []VersionedConfigWithDigest{primaryConfig, secondaryConfig}
+			// Create a variable to track if GetLatestValue was called
+			var getLatestValueCalled bool
+			var mu sync.Mutex
 
-	homeChainReader.On(
-		"GetLatestValue",
-		mock.Anything,
-		mock.Anything,
-		mock.Anything,
-		mock.Anything,
-		mock.Anything,
-	).Run(func(args mock.Arguments) {
-		arg := args.Get(4).(*[]VersionedConfigWithDigest)
-		*arg = rmnHomeOnChainConfigs
-	}).Return(nil).Times(50)
+			homeChainReader.On("GetLatestValue",
+				mock.Anything,
+				mock.Anything,
+				mock.Anything,
+				mock.Anything,
+				mock.Anything,
+			).Run(func(args mock.Arguments) {
+				result := args.Get(4).(*[]VersionedConfigWithDigest)
+				*result = []VersionedConfigWithDigest{
+					{
+						ConfigDigest: [32]byte{1},
+						VersionedConfig: VersionedConfig{
+							Version: 1,
+							Config: Config{
+								Nodes:        []Node{},
+								SourceChains: []SourceChain{},
+							},
+						},
+					},
+				}
+				mu.Lock()
+				getLatestValueCalled = true
+				mu.Unlock()
```

Comment:
Why is this needed? The mock can tell you if its called or not - if its called, the test passes, if not, it fails. Does this do anything different?